examples/lighting-app/k32w/main/AppTask.cpp

```diff
@@ -0,0 +1,561 @@
+    {
+        action = static_cast<LightingManager::Action_t>(aEvent->LightEvent.Action);
+        actor  = aEvent->LightEvent.Actor;
+    }
+    else if (aEvent->Type == AppEvent::kEventType_Button)
+    {
+        if (LightingMgr().IsTurnedOff())
+        {
+            action = LightingManager::TURNON_ACTION;
+        }
+        else
+        {
+            action = LightingManager::TURNOFF_ACTION;
+        }
+    }
+    else
+    {
+        err = CHIP_ERROR_MAX;
+    }
+
+    if (err == CHIP_NO_ERROR)
+    {
+        initiated = LightingMgr().InitiateAction(actor, action);
+
+        if (!initiated)
+        {
+            K32W_LOG("Action is already in progress or active.");
+        }
+    }
+}
+
+void AppTask::ThreadStart()
+{
+    chip::DeviceLayer::Internal::DeviceNetworkInfo networkInfo;
+
+    memset(networkInfo.ThreadNetworkName, 0, chip::DeviceLayer::Internal::kMaxThreadNetworkNameLength + 1);
+    memcpy(networkInfo.ThreadNetworkName, "OpenThread", 10);
+
+    networkInfo.ThreadExtendedPANId[0] = 0xde;
+    networkInfo.ThreadExtendedPANId[1] = 0xad;
+    networkInfo.ThreadExtendedPANId[2] = 0x00;
+    networkInfo.ThreadExtendedPANId[3] = 0xbe;
+    networkInfo.ThreadExtendedPANId[4] = 0xef;
+    networkInfo.ThreadExtendedPANId[5] = 0x00;
+    networkInfo.ThreadExtendedPANId[6] = 0xca;
+    networkInfo.ThreadExtendedPANId[7] = 0xfe;
+
+    networkInfo.ThreadMasterKey[0]  = 0x00;
+    networkInfo.ThreadMasterKey[1]  = 0x11;
+    networkInfo.ThreadMasterKey[2]  = 0x22;
+    networkInfo.ThreadMasterKey[3]  = 0x33;
+    networkInfo.ThreadMasterKey[4]  = 0x44;
+    networkInfo.ThreadMasterKey[5]  = 0x55;
+    networkInfo.ThreadMasterKey[6]  = 0x66;
+    networkInfo.ThreadMasterKey[7]  = 0x77;
+    networkInfo.ThreadMasterKey[8]  = 0x88;
+    networkInfo.ThreadMasterKey[9]  = 0x99;
+    networkInfo.ThreadMasterKey[10] = 0xAA;
+    networkInfo.ThreadMasterKey[11] = 0xBB;
+    networkInfo.ThreadMasterKey[12] = 0xCC;
+    networkInfo.ThreadMasterKey[13] = 0xDD;
+    networkInfo.ThreadMasterKey[14] = 0xEE;
+    networkInfo.ThreadMasterKey[15] = 0xFF;
+
+    networkInfo.ThreadPANId   = 0xabcd;
+    networkInfo.ThreadChannel = 15;
+
+    networkInfo.FieldPresent.ThreadExtendedPANId = true;
+    networkInfo.FieldPresent.ThreadMeshPrefix    = false;
+    networkInfo.FieldPresent.ThreadPSKc          = false;
+    networkInfo.NetworkId                        = 0;
+    networkInfo.FieldPresent.NetworkId           = true;
+
+    ThreadStackMgr().SetThreadEnabled(false);
+    ThreadStackMgr().SetThreadProvision(networkInfo);
+    ThreadStackMgr().SetThreadEnabled(true);
+}
+
+void AppTask::JoinHandler(AppEvent * aEvent)
+{
+    if (aEvent->ButtonEvent.PinNo != JOIN_BUTTON)
+        return;
+
+    if (sAppTask.mFunction != kFunction_NoneSelected)
+    {
+        K32W_LOG("Another function is scheduled. Could not initiate Thread Join!");
+        return;
+    }
+
+    /* hard-code Thread Commissioning Parameters for the moment.
```

Comment:
Do we need this? Setting the thread dataset via K32W isn't working yet? Why not?

Reply:
The user has the option of pushing one of the buttons on the K32W and this will trigger a Thread join that it's using the above hard-coded parameters.